opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractParentSpanResolutionTest.java

```diff
@@ -1,26 +1,49 @@
 package io.opentracing.contrib.jaxrs2.itest.common;
 
-import static org.awaitility.Awaitility.await;
-
 import io.opentracing.Scope;
+import io.opentracing.contrib.jaxrs2.client.ClientTracingFeature;
+import io.opentracing.contrib.jaxrs2.server.ServerTracingDynamicFeature;
+import io.opentracing.contrib.jaxrs2.server.SpanFinishingFilter;
 import io.opentracing.mock.MockSpan;
 import io.opentracing.tag.Tags;
 import org.eclipse.jetty.servlet.FilterHolder;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 import org.junit.Assert;
 import org.junit.Test;
 
-import java.io.IOException;
-import java.util.EnumSet;
-import java.util.List;
-
 import javax.servlet.*;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.List;
+
+import static org.awaitility.Awaitility.await;
 
 public abstract class AbstractParentSpanResolutionTest extends AbstractJettyTest {
 
+    protected abstract boolean shouldUseParentSpan();
+
+    @Override
+    protected void initTracing(ServletContextHandler context) {
```

Comment:
This method is copied from the `AbstractServerDefaultConfigurationTest`, but with some logic to control whether or not we want to use the new `withJoinExistingActiveSpan` method in the builder.